Server/src/main/java/org/xdi/oxauth/auth/AuthenticationFilter.java

```diff
@@ -130,7 +130,7 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
                 if (StringUtils.isNotBlank(sessionId)) {
                     sessionIdObject = sessionIdService.getSessionId(sessionId);
                 }
-                if (sessionIdObject != null && SessionIdState.AUTHENTICATED == sessionIdObject.getState()
+                if (sessionIdObject != null && !StringUtils.isEmpty(sessionIdObject.getState().get("state"))

```

Comment:
@rajnikantsh how `AUTHENTICATED` state of SessionId is related to attribute stored inside state ? Would you please explain what exactly are u trying to achieve here ?

Reply:
@yuriyz I was doing this to replace state token with json object that can store the state token along with other parameters. such as state = {"idp": "custa", "state": "72DC91332Fskfjdk"}.  for this i replaced the state with map to store key value pair values.

Reply:
@rajnikantsh 
This line checks whether session id state is authenticated
```
SessionIdState.AUTHENTICATED == sessionIdObject.getState()
```
You replaced it with this line
```
!StringUtils.isEmpty(sessionIdObject.getState().get("state"))
```
which checks whether opaque `state` value string is not empty. I means you removed check whether session is authenticated or not. As result `processSessionAuth` will not be invoked if session is AUTHENTICATED and even authenticated user will not be able to get the token by session. From other side even NOT authenticated user IS able to get token, then only thing he/she needs it to provide `state` parameter which is not blank. So question why do we process session authentication for unauthenticated session. Is not empty `state` parameter enough for it ?

Reply:
@yuriyz thats the mistake from my end. In future I will make sure to properly consider all aspects for critical issues like this. As i mentioned before i thought i could replace token with json value easily to have state token and didn't realize it would break whole authentication flow.